sympy/stats/tests/test_rv.py

```diff
@@ -153,6 +153,15 @@ def test_random_symbol_no_pspace():
     x = RandomSymbol(Symbol('x'))
     assert x.pspace == PSpace()
 
+def test_iid():
+    raises(TypeError, lambda: iid("I", Normal("N", 0, 1)))
+    raises(ValueError, lambda: iid("I", Normal("N", 0, 1), -1))
+
+    I = iid("I", Normal("N", 0, 1), 5)
```

Comment:
This is a strang way to define a variable. Why would you need to nest a random symbol _N_ inside another random symbol? You specify another variable _"I"_ up the expression tree.

What about not specifying _"I"_?

Reply:
The idea is that `I` is a collection of 5 independent and identically distributed variables having a Normal distribution. Then, we could refer to the nth RV in the collection as `I[n]` and nth order statistic as `I(n)`. Since users can't use distribution classes directly, I did this. The nesting is only a way to get the desired distribution. What interface do you propose apart from this?

Reply:
How is it different from making a simple list of `Normal` objects?

Reply:
It isn't, I just thought we could stay true to the notation commonly used. And maybe this would be more efficient if no. of RVs increases

Reply:
what about not introducing `iid` at all? What about this syntax:
```python
I = Normal("I", 0, 1, repeat_iid=5)
```
instead of
```python
I = iid("I", Normal("N", 0, 1), 5)
```

Reply:
Consider that you can already create `iid` of normal distributions with:
```python
I = Normal("I", [0, 0, 0, 0, 0],  [[1, 0, 0, 0, 0], [0, 1, 0, 0, 0], [0, 0, 1, 0, 0], [0, 0, 0, 1, 0], [0, 0, 0, 0, 1]])
```
This statement is meant to handle dependent RVs (in this case a multivariate gaussian), but in this case it creates IID RVs, as the covariance is diagonal, and all means and variances are the same.

As you can see, we just use `Normal` to create both dependent and independent RVs... I don't see why we should make everything overly complicated with a new statement like `iid` to use in such a way as it has been proposed.

Reply:
> This statement is meant to handle dependent RVs (in this case a multivariate gaussian), but in this case it creates IID RVs, as the covariance is diagonal, and all means and variances are the same.

Yes, but what about other distributions like Poisson/Beta? I don't know if there is a way to do this for all distributions

Reply:
I'm actually trying to find a simpler API design for IID RVs. I would prefer to find a simple way to define IID RVs.

The more complicated you make the API, the less people will use SymPy. Consider that SymPy is already quite hard to learn.

Reply:
Definitely, I'm not saying we have to use a separate class. We could discuss more and come up with a simpler solution